apps/studio/pages/api/edge-functions/test.ts

```diff
@@ -21,6 +21,19 @@ async function handlePost(req: NextApiRequest, res: NextApiResponse) {
   try {
     const { url, method, body: requestBody, headers: customHeaders } = req.body
 
+    const regexValidEdgeFunctionURL = new RegExp(
+      'https://[a-z]*.supabase.(red|co)/functions/v[0-9]{1}/S*',
```

Comment:
```suggestion
      '^https://[a-z]*.supabase.(red|co)/functions/v[0-9]{1}/\S*$',
```

Anchoring the regex to the start of the string, otherwise it leaves room for something like `https://localhost/?https://aaaa.supabase.co/functions/v1/xxx`

And I think the `/S*` should be `/\S*` to capture the intent Any non-whitespace character, otherwise it is matching the character `S` zero or more times 

For good measure we can have the end anchor in there as well (RegExp probably also has match the whole string, rather than test - can't recall)

Reply:
awesome thank you so much for checking!! 🙏 🙏